auction-app-backend/src/main/java/com/example/auctionapp/entity/ProductEntity.java

```diff
@@ -109,33 +110,30 @@ public ProductEntity(final UUID productId,
     }
 
     public Product toDomainModel() {
-        Product product = new Product();
-
-        product.setId(this.productId);
-        product.setName(this.name);
-        product.setDescription(this.description);
-        product.setStartPrice(this.startPrice);
-        product.setStartDate(this.startDate);
-        product.setEndDate(this.endDate);
-        product.setStatus(this.status);
-        product.setCategoryId(this.categoryEntity.getCategoryId());
+        GenericBuilder<Product> builder = GenericBuilder.of(Product::new)
```

Comment:
Can this be final?

Reply:
i couldnt make it final because i assign a value to builder below, inside the if block